background.js

```diff
@@ -32,8 +32,22 @@ function ssoLogError(message) {
     console.error("[Linux Entra SSO] " + message);
 }
 
-function isFirefox() {
-    return typeof browser !== "undefined";
+function getBrowser() {
+    let userAgent = navigator.userAgent.toLowerCase();
+
+    if (userAgent.includes("firefox")) {
+        return "Firefox";
+    } else if (userAgent.includes("thunderbird")) {
+        return "Thunderbird";
+    } else if (userAgent.includes("chrome")) {
+        return "Chrome";
+    } else {
+        return "Unknown";
+    }
```

Comment:
How does a user agent switcher interferes with this?

Reply:
I started major rework (and cleanup) to properly support platform specific code in #78 . This also makes the extension more modular, which helps to implement proper Thunerbird support.